src/Agoda.Analyzers/AgodaCustom/AG0019PreventUseOfInternalsVisibleToAttribute.cs

```diff
@@ -34,14 +35,17 @@ public AG0019PreventUseOfInternalsVisibleToAttribute()
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(_diagnosticDescriptor);
 
-        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.AttributeList);
+        public override void Initialize(AnalysisContext context) =>
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.AttributeList);
 
         private void AnalyzeNode(SyntaxNodeAnalysisContext context)
         {
             var attributeNode = (AttributeListSyntax)context.Node;
-
-            if (attributeNode.Target?.Identifier.Text != "assembly") return;
-
+            var targetSyntaxTree = context.Compilation?.SyntaxTrees.FirstOrDefault(tree => tree == attributeNode.SyntaxTree);
+            if (attributeNode.Target?.Identifier.Text != "assembly" || targetSyntaxTree == null || targetSyntaxTree.FilePath.ToLower().Contains("test"))
```

Comment:
I think we should do EndsWith Test or Tests as some words contain test, such as "Contest"